tests/lib/fakestore/store/store.go

```diff
@@ -739,49 +740,101 @@ func (s *Store) retrieveAssertion(bs asserts.Backstore, assertType *asserts.Asse
 	return a, err
 }
 
-func (s *Store) assertionsEndpoint(w http.ResponseWriter, req *http.Request) {
-	assertPath := strings.TrimPrefix(req.URL.Path, "/v2/assertions/")
+func (s *Store) retrieveLatestSequenceFormingAssertion(bs asserts.Backstore, assertType *asserts.AssertionType, sequenceKey []string) (asserts.Assertion, error) {
+	a, err := bs.SequenceMemberAfter(assertType, sequenceKey, -1, assertType.MaxSupportedFormat())
+	if errors.Is(err, &asserts.NotFoundError{}) && s.assertFallback {
+		return s.fallback.SeqFormingAssertion(assertType, sequenceKey, -1, nil)
+	}
+	return a, err
+}
 
-	bs, err := s.collectAssertions()
-	if err != nil {
-		http.Error(w, fmt.Sprintf("internal error collecting assertions: %v", err), 500)
-		return
+func (s *Store) sequenceFromQueryValues(values url.Values) (int, error) {
+	if val, ok := values["sequence"]; ok {
+		// special case value of 'latest', in that case
+		// we return -1 to indicate we want the newest
+		if val[0] != "latest" {
+			seq, err := strconv.Atoi(val[0])
+			if err != nil {
+				return -1, fmt.Errorf("cannot parse sequence %s: %v", val[0], err)
+			}
+			// XXX: Treat values 0 and below as error? Anyone know the official store
+			// behavior on this?
```

Comment:
I think it errors:

sequence param must be a positive integer or "latest"

Reply:
Thank you for the insight, changed this and the unit test to error on 0 and negative values